ui/src/services/axiosInstances.ts

```diff
@@ -22,21 +22,41 @@
 
 import axios from 'axios'
 
+const detectAuthHeaders = () => {
+  if (import.meta.env.VITE_BASE_USER && import.meta.env.VITE_BASE_PASSWORD) {
+    const user = import.meta.env.VITE_BASE_USER.toString()
+    const password = import.meta.env.VITE_BASE_PASSWORD.toString()
+    const credentials = `${user}:${password}`
+    return {
+      'Authorization': `Basic ${btoa(credentials)}`
+    }
+  }
+  return null;
+}
+const authHeaders = detectAuthHeaders();
+
 const v2 = axios.create({
   baseURL: import.meta.env.VITE_BASE_V2_URL?.toString() || '/opennms/api/v2',
-  withCredentials: true
+  withCredentials: authHeaders != null,
```

Comment:
I think we need to leave this as `withCredentials: true`, it's needed for normal non-vite-server usage. It makes axios send the JWT token, etc.

Reply:
Is this re-used somewhere? Because the only difference here is, that it does not send the cookie when in "dev mode".

Reply:
If I'm reading this correctly, `detectAuthHeaders()` will be false if either `VITE_BASE_USER` or `VITE_BASE_PASSWORD` are not defined. These are only defined in `.env.development`, so won't be present during production. So in production, `withCredentials` will be set to `false` which we don't want.

In addition, we should have some additional fail safe check like:

```
withCredentials: import.meta.env.PROD || authHeaders != null
```

I may be missing something, but want to make sure this isn't a security hole.

Reply:
I can take another spin at this at some point. Maybe we can just leave it as is and it will work nevertheless. Then we don't have to worry about it.

Just as a side note, it should be `authHeaders == null`, as otherwise it is definitely wrong

Reply:
@mvrueden I think that last part was really throwing me off. I'll be happy to approve this with 2 changes:

- update that line to `withCredentials: import.meta.env.PROD || authHeaders === null`
- add a comment in `detectAuthHeaders()`, something like this below, just to make the intention more clear: 

> `VITE_BASE_USER`  and `VITE_BASE_PASSWORD` should only be defined in development mode, not in production

Also, make sure to test in dev and prod mode and make sure this works in either case. This would be a nice thing to have for improving the development experience.